src/resolvers/default.rs

```diff
@@ -452,48 +451,53 @@ impl Hash for HashBLAKE2s {
     }
 }
 
-#[cfg(feature = "pqclean_kyber1024")]
+#[cfg(feature = "oqs_kyber1024")]
 impl Default for Kyber1024 {
     fn default() -> Self {
+        let kemalg = kem::Kem::new(kem::Algorithm::Kyber1024).unwrap();
+        let (pk, sk) = kemalg.keypair().unwrap();
         Kyber1024 {
-            pubkey:  kyber1024::PublicKey::from_bytes(&[0; kyber1024::public_key_bytes()]).unwrap(),
-            privkey: kyber1024::SecretKey::from_bytes(&[0; kyber1024::secret_key_bytes()]).unwrap(),
+            pubkey:  pk,
+            privkey: sk,
+            alg: kemalg,
         }
     }
 }
 
-#[cfg(feature = "pqclean_kyber1024")]
+#[cfg(feature = "oqs_kyber1024")]
 impl Kem for Kyber1024 {
     fn name(&self) -> &'static str {
         "Kyber1024"
     }
 
     /// The length in bytes of a public key for this primitive.
     fn pub_len(&self) -> usize {
-        kyber1024::public_key_bytes()
+        self.alg.length_public_key()
     }
 
     /// The length in bytes the Kem cipherthext for this primitive.
     fn ciphertext_len(&self) -> usize {
-        kyber1024::ciphertext_bytes()
+        self.alg.length_ciphertext()
     }
 
     /// Shared secret length in bytes that this Kem encapsulates.
     fn shared_secret_len(&self) -> usize {
-        kyber1024::shared_secret_bytes()
+        self.alg.length_shared_secret()
     }
 
     /// Generate a new private key.
     fn generate(&mut self, _rng: &mut dyn Random) {
-        // PQClean uses their own random generator
-        let (pk, sk) = kyber1024::keypair();
+        // Oqs uses their own random generator
+        let kemalg = kem::Kem::new(kem::Algorithm::Kyber1024).unwrap();
```

Comment:
Is it necessary to create a new `Kem` every time you call `generate()`?

Reply:
ah good point. i think it is not necessary to create it each time generate is called.